File: Difficulty: Medium/Find H-Index/find-hindex.cpp
class Solution {
  public:
    int hIndex(vector<int>& citations) {
        int n=citations.size();
        
        priority_queue<int> pq;
        
        for(auto& it: citations)pq.push(it); 
        int cnt=0;
        
        int ans=-1e9;
        int maxi=1e9;
        while(!pq.empty())
        {
            int a=pq.top();
            pq.pop();
            cnt++;
            maxi=min(maxi,a);
            if(maxi>=cnt)
            {
                ans=max(ans,cnt);
            }
        }
        if(ans==-1e9)return 0;
        return ans;
    }
};

File: Difficulty: Medium/Find H-Index/find-hindex.java
class Solution {
    public int hIndex(int[] citations) {
        int start=1,end=citations.length;
        while(start<=end) {
            int mid = start+(end-start)/2;
            if(check(mid,citations)) {
                start = mid+1;
            }
            else {
                end=mid-1;
            }
        }
        return end;
    }
    public boolean check(int num,int[] arr) {
        int count=0;
        for(int i=0;i<arr.length;i++) {
            if(arr[i]>=num) {
                count++;
            }
        }
        return count>=num;
    }
}